src/main/scheduler/scheduler_tasks.c

```diff
@@ -31,13 +31,34 @@ cfTask_t cfTasks[TASK_COUNT] = {
         .staticPriority = TASK_PRIORITY_HIGH,
     },
 
-    [TASK_GYROPID] = {
-        .taskName = "GYRO/PID",
-        .taskFunc = taskMainPidLoopChecker,
-        .desiredPeriod = 1000,
+    [TASK_PID] = {
+        .taskName = "PID",
+        .taskFunc = taskMainPidLoop,
+        .desiredPeriod = 1000000 / 500, // Run at 500Hz
         .staticPriority = TASK_PRIORITY_REALTIME,
     },
 
+    [TASK_GYRO] = {
+        .taskName = "GYRO",
+        .taskFunc = taskGyro,
+        .desiredPeriod = 1000000 / 1000, //Run at 1000Hz
+        .staticPriority = TASK_PRIORITY_REALTIME,
```

Comment:
Current implementation of scheduler assumes only one task of TASK_PRIORITY_REALTIME priority, otherwise timely execution is not guaranteed. IMO TASK_PID should be of HIGH priority, not realtime.